app/MMB/src/navigation/DetailScreen.js

```diff
@@ -166,3 +171,21 @@ export default class DetailsScreen extends React.Component<Props, State> {
     );
   };
 }
+
+type Props = {|
+  isPastBooking: boolean,
+  bookingId: string,
+  navigation: NavigationType,
+|};
+
+export default function DetailScreenWithContext(props: Props) {
+  return (
+    <BookingDetailContext.Provider>
+      <BookingDetailContext.Consumer>
```

Comment:
This looks odd. Don't be afraid to use the `Consumer` directly in the component...

Reply:
Then how would I inject the function needed to set the property?